apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/McsClaimResponseTransformerV2.java

```diff
@@ -96,11 +97,22 @@ private static ClaimResponse transformClaim(PreAdjMcsClaim claimGroup) {
     claim.setContained(List.of(getContainedPatient(claimGroup)));
     claim.setExtension(getExtension(claimGroup));
     claim.setIdentifier(getIdentifier(claimGroup));
-    claim.setStatus(
-        CANCELLED_STATUS_CODES.contains(claimGroup.getIdrStatusCode().toLowerCase())
-            ? ClaimResponse.ClaimResponseStatus.CANCELLED
-            : ClaimResponse.ClaimResponseStatus.ACTIVE);
-    claim.setOutcome(OUTCOME_MAP.get(claimGroup.getIdrStatusCode()));
+
+    // Some status codes for MCS can be null.
+    // These will be interpreted as status=active, outcome=queued.
+    if (claimGroup.getIdrStatusCode() == null) {
+      claim.setStatus(ClaimResponse.ClaimResponseStatus.ACTIVE);
+    } else {
+      claim.setStatus(
+          CANCELLED_STATUS_CODES.contains(claimGroup.getIdrStatusCode().toLowerCase())
+              ? ClaimResponse.ClaimResponseStatus.CANCELLED
+              : ClaimResponse.ClaimResponseStatus.ACTIVE);
+    }
+
+    claim.setOutcome(
+        ObjectUtils.defaultIfNull(
+            OUTCOME_MAP.get(ObjectUtils.defaultIfNull(claimGroup.getIdrStatusCode(), "")),
+            ClaimResponse.RemittanceOutcome.QUEUED));
```

Comment:
This might be a little more direct:

```
    // Some status codes for MCS can be null.
    // These will be interpreted as status=active, outcome=queued.
    if (claimGroup.getIdrStatusCode() == null) {
      claim.setStatus(ClaimResponse.ClaimResponseStatus.ACTIVE);
      claim.setOutcome(ClaimResponse.RemittanceOutcome.QUEUED);
    } else {
      String statusCode = Strings.nullToEmpty(claimGroup.getIdrStatusCode()).toLowerCase();
      claim.setStatus(
          CANCELLED_STATUS_CODES.contains(statusCode)
              ? ClaimResponse.ClaimResponseStatus.CANCELLED
              : ClaimResponse.ClaimResponseStatus.ACTIVE);
      claim.setOutcome(
          OUTCOME_MAP.getOrDefault(statusCode, ClaimResponse.RemittanceOutcome.QUEUED));
    }
```

Making CANCELLED_STATUS_CODES a Map would let you eliminate the ternary operator in `setStatus` too since you could just use `getOrDefault` there.

Reply:
Good point about the extra `nullToEmpty()`.  My mind. was on the previous comment about using `Optional`.

I do think using a `STATUS_CODES_MAP` instead of `CANCELLED_STATUS_CODES` set would make the code clearer though.  It would let the code adapt other statuses later too.